src/services/command/command.ts

```diff
@@ -8,7 +8,12 @@ import { relativeToThisFile } from "../util/paths.js";
 import type { ArgsSpec } from "./arg.js";
 import type { Context } from "./context.js";
 
-export const AvailableCommands = ["sync", "list", "login", "logout", "whoami", "version"] as const;
+export const AvailableCommands = ["sync", "deploy", "list", "login", "logout", "whoami", "version"] as const;
+
+// temporarily remove deploy command from available commands while we're working on it
+if (!process.env["GGT_DEPLOY_PREVIEW"]) {
+  (AvailableCommands as unknown as string[]).splice(AvailableCommands.indexOf("deploy"), 1);
+}
```

Comment:
Placing `deploy` inside the list of commands by default will make it appear in the README because it's used [here](https://github.com/gadget-inc/ggt/blob/main/scripts/generate-readme.ts#L30).

If we don't care about that, then all good 👍 -- but if we don't want it in the README, we can do the opposite and only add it to the array if the env var is set.

```ts
if (process.env["GGT_DEPLOY_PREVIEW"]) {
  (AvailableCommands as unknown as string[]).push("deploy");
}
```

Reply:
Won't that script get the modified value when it executes as well, because the env var won't be set?

I originally tried to do it like you suggested as that makes more sense to me, but then it being absent from the type (because of `as const`) caused a bunch of type errors later.

Reply:
`GGT_DEPLOY_PREVIEW` won't be set when the script runs, so it shouldn't end up in the README.

---

I just copy/pasted my suggestion and didn't run into any type errors. Do you know which errors you ran into?

```ts
if (process.env["GGT_DEPLOY_PREVIEW"]) {
  (AvailableCommands as unknown as string[]).push("deploy");
}
```

Reply:
Hm I must have had something else broken, you're right! Updated.